src/qml/facebook/facebookimagecachemodel.cpp

```diff
@@ -296,7 +298,11 @@ void FacebookImageCacheModel::queryFinished()
             //% "All"
             albumMap.insert(FacebookImageCacheModel::Title, qtTrId("nemo_socialcache_facebook_images_model-all-albums"));
             albumMap.insert(FacebookImageCacheModel::Count, count);
-            albumMap.insert(FacebookImageCacheModel::UserId, fbUserId);
+            if (d->nodeIdentifier.isEmpty()) {
+                albumMap.insert(FacebookImageCacheModel::UserId, QString());
+            } else {
+                albumMap.insert(FacebookImageCacheModel::UserId, data.first().value(FacebookImageCacheModel::UserId));
```

Comment:
albumData doesn't exist any more at this point. I could use a helper variable, as before, but I felt this was more efficient since that variable is not always needed and accessing the value is quite straight forward.